### PR TITLE
chore(gui-client/linux): poll for DNS changes every 5 seconds

### LINKED ISSUE #4819: linux Client: Network connectivity / DNS change events

### PR DESCRIPTION
**Pull request by ReactorScram**
Closes #4819

There is a way to get them async with D-Bus. We can invest time in that if we want. Polling works for now, it's just gonna be a slight battery waste on laptops.

### CODE REVIEW

File: rust/gui-client/src-tauri/src/client/network_changes/linux.rs [hunk 1]
```diff
@@ -17,29 +19,57 @@ pub(crate) fn check_internet() -> Result<bool> {
...
     pub(crate) fn close(&mut self) -> Result<()> {
         Ok(())
     }
 
-    pub(crate) async fn notified(&self) {
-        futures::future::pending().await
+    pub(crate) async fn notified(&mut self) {
+        loop {
+            self.interval.tick().await;
+            tracing::trace!("Checking for network changes");
```
Comment (ReactorScram) [inline rust/gui-client/src-tauri/src/client/network_changes/linux.rs]: I didn't implement this one, but this is where the Windows Client watches for loss / gain of Internet.

Comment (jamilbk) [reply rust/gui-client/src-tauri/src/client/network_changes/linux.rs]: is it worth adding a note/comment here?

File: rust/gui-client/src-tauri/src/client/network_changes/linux.rs [hunk 2]
```diff
@@ -17,29 +19,57 @@ pub(crate) fn check_internet() -> Result<bool> {
...
     }
-    pub(crate) async fn notified(&mut self) -> Result<()> {
-        futures::future::pending().await
+
+    pub(crate) async fn notified(&mut self) -> Result<Vec<IpAddr>> {
+        loop {
+            self.interval.tick().await;
+            tracing::trace!("Checking for DNS changes");
+            let new = crate::client::resolvers::get().unwrap_or_default();
+            if new != self.last_seen {
```
Comment (jamilbk) [inline rust/gui-client/src-tauri/src/client/network_changes/linux.rs:59]: Might help to have a debug log for this

```suggestion
            if new != self.last_seen {
                tracing::debug!("Found new DNS resolvers: {:?}", new);
```

**Review by jamilbk — APPROVED** [review]
minor suggestion

File: rust/gui-client/src-tauri/src/client/network_changes/linux.rs (hunk 1 above)
Comment (ReactorScram) [reply rust/gui-client/src-tauri/src/client/network_changes/linux.rs]: Yeah I'll add a note and just change it back to an infinite `future::pending`

File: rust/gui-client/src-tauri/src/client/network_changes/linux.rs (hunk 2 above)
Comment (ReactorScram) [reply rust/gui-client/src-tauri/src/client/network_changes/linux.rs:59]: Good idea. There's already one in the platform-independent code in `gui.rs` so I added the list of resolvers there.

**Pull request merged**